### DISCUSSION TITLE
Network configuration for clients

### DISCUSSION THREAD

**Discussion by SynchronicIT** [edgexfoundry/community#198] [Q&A]
Hi all, 

We are using EdgeX for Data analytics and measurement tools. These are in most cases not permanently installed. 
For now we are running these on raspberry pi 4 boards, but we want to move to compute modules or similar headless solutions. 

How do you manage the network settings of the gateways, or what do you provide to your customers to configure the network settings (ethernet, wifi, ... whatever is applicable). Something off-the-shelf console panel, or custom / in house made? 
I've compared several readily made console panels (webmin, Ajenti, cockpit, etc. etc.) but non of them fit the shoe. 

Thanks
Vincent

**Comment by cloudxxx8** [comment]
The network settings should really depends on the use case and the physical environment.  If the ethernet cable connection is available, it is better than wifi for sure.

**Reply by SynchronicIT** [reply]
what I meant is, how to you allow customers to configure the network settings on your edge gateways, without giving them access to ssh or admin rights? 
We made our own dashboard + wifi access point method, but I find it strange no standard design exists for this within edgeX. Everything is pretty much thought of, besides the tools to actually being able to access the edge gateway.

**Reply by cloudxxx8** [reply]
EdgeX is the application level software, so the users could decide their own network settings and the configuration approach.  We don't want to restrict anything in the OS or hardware level.

**Reply by SynchronicIT** [reply]
Of course, it's not something which has to be an integral part of edgeX. But as most of us have to handle this, I was wondering how others do this.
